packages/api/__tests__/headers.test.ts

```diff
@@ -0,0 +1,224 @@
+import { Api } from '../src';
+import { IApiTestsListData } from './types';
+
+const { TEST_WORKSPACE_ID } = process.env;
+
+describe('headers', () => {
+  let originalFetch;
+
+  beforeAll(() => {
+    originalFetch = fetch;
+    // @ts-ignore
+    fetch = jest.fn((...args) => originalFetch(...args));
+  });
+
+  afterEach(() => {
+    // @ts-ignore
+    fetch.mockClear();
+  });
+
+  afterAll(() => {
+    // @ts-ignore
+    fetch = originalFetch;
+  });
+
+  it('passes headers from constructor. Object', () => {
+    const api = new Api({
+      workspaceId: TEST_WORKSPACE_ID,
+      headers: {
+        auth: 'Some token',
+      },
+    });
+
+    api.request<IApiTestsListData>(`
+      query {
+        apiTestsList {
+          items {
+            id
+            timestamp
+          }
+        }
+      }
+    `);
+
+    expect(fetch).lastCalledWith(
+      expect.any(String),
+      expect.objectContaining({
+        body: expect.any(String),
+        method: expect.any(String),
+        headers: {
+          auth: 'Some token',
+          'content-type': 'application/json',
+        },
+      }),
+    );
+  });
+
+  it('passes headers from constructor. Function', () => {
+    const api = new Api({
+      workspaceId: TEST_WORKSPACE_ID,
+      headers: () => ({
+        auth: 'Some token',
+      }),
+    });
+
+    api.request<IApiTestsListData>(`
+      query {
+        apiTestsList {
+          items {
+            id
+            timestamp
+          }
+        }
+      }
+    `);
+
+    expect(fetch).lastCalledWith(
+      expect.any(String),
+      expect.objectContaining({
+        body: expect.any(String),
+        method: expect.any(String),
+        headers: {
+          auth: 'Some token',
+          'content-type': 'application/json',
+        },
+      }),
+    );
+  });
+
+  it('passes headers from method', () => {
```

Comment:
I believe that this makes it necessary as well to follow to normal "Authentication": "Bearer TOKEN" convention, as we wouldn't coerce the headers passed to the method, and wouldn't want to give 2 different declaration styles for setting the token/passing headers to the method.

Reply:
I don't understand. 
In some cases you need to pass some custom headers just for one request or rewrite general headers described in constructor. (I.e. you want to do not pass the token for a `signUp` mutation). 

What's wrong with that?

Reply:
Misunderstanding from the Auth key in header test. No problems.